components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesHistoryStorage.kt

```diff
@@ -146,4 +149,21 @@ open class PlacesHistoryStorage(context: Context) : PlacesStorage(context), Hist
             places.writer().pruneDestructively()
         }
     }
+
+    /**
+     * Runs syncHistory() method on the places Connection
+     *
+     * @param authInfo The authentication information to sync with.
+     * @return Sync status of OK or Error
+     */
+    override suspend fun sync(authInfo: AuthInfo): SyncStatus {
+        return try {
+            withContext(scope.coroutineContext) {
+                places.syncHistory(authInfo.into())
+                SyncStatus.Ok
+            }
+        } catch (e: PlacesException) {
```

Comment:
Nit: could you report (or fail to catch, assuming something else catches and reports) rust panics here (and above at the bookmark sync)

Reply:
This will be reported upstream.